cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -183,9 +186,9 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
                 .compose(state -> state.zkServiceEndpointReadiness())
                 .compose(state -> state.zkHeadlessServiceEndpointReadiness())
                 .compose(state -> state.zkPersistentClaimDeletion())
-                .compose(state -> state.kafkaUpgrade())
                 .compose(state -> state.kafkaManualPodCleaning())
                 .compose(state -> state.kafkaManualRollingUpdate())
+                .compose(state -> state.kafkaUpgrade())
```

Comment:
Assuming we fix the waiting for readiness, do we still need to change the order?

Reply:
Maybe not